### ISSUE TITLE
[BUG] `FeatureFlags.isEnabled` is kinda slow

### ISSUE THREAD

**Issue by andrross** [opensearch-project/OpenSearch#16519]
### Describe the bug

After having run the full `big5` benchmark to create the index, I ran the following command to isolate a simple query and run it as fast as possible:

```
opensearch-benchmark execute-test \
  --target-hosts localhost:9200 \
  --workload big5 --workload-params 'number_of_replicas:0,target_throughput:0,search_clients:8,test_iterations:100000' \
  --pipeline benchmark-only --include-tasks term \
  --kill-running-processes
```

This is a super fast query, so I was specifically looking for overhead in things not related to doing heavyweight search work. It turns out `FeatureFlags.isEnabled` is taking almost 5% of CPU cycles. It is unsurprising that a feature flag check might pop up in hot path code, and there's no reason that looking up a setting value should be so expensive. It turns out the `System.getProperty` check is kind of slow (because of a security manager check it seems).

### Related component

Search

### To Reproduce

See above

### Expected behavior

Features flags defined in system properties are not dynamic. All feature flags are statically known. We should pre-populate these booleans into an immutable map during static initialization to avoid the `System.getProperty` call at runtime.

### Additional Details

<img width="1634" alt="image" src="https://github.com/user-attachments/assets/6be03b06-8f59-41c1-ac67-1b37ec61bb70">

**Comment by msfroh** [comment]
A comment was added as part of https://github.com/opensearch-project/OpenSearch/pull/4959 suggesting that we remove that `if` statement once feature flags can only be set via opensearch.yml.

I think we may need to fix `FeatureFlagSetter`, which uses system properties to dynamically update feature flags in unit tests.

**Comment by andrross** [comment]
I think `FeatureFlagSetter` can probably be removed and replaced with calls to `FeatureFlags.initializeFeatureFlags`, which seems to be more widely used in tests.

**Comment by msfroh** [comment]
Yeah, that makes sense. We can clear the flags in `OpenSearchTestCase#tearDown()` using `FeatureFlags.initializeFeatureFlags`. The automatic clearing of feature flags on teardown was arguably the most useful part of `FeatureFlagSetter`.

**Comment by bhngupta** [comment]
This seems interesting, can I take this up?

**Comment by dbwiddis** [comment]
> This seems interesting, can I take this up?

In the Open Source world the answer is almost always, universally, an all-caps-shouted YES!

Assigning to you!

**Comment by sandeshkr419** [comment]
[Search Triage] @bhngupta Hi, were you able to get started on this?

**Comment by finnegancarroll** [comment]
Taking a look at this.
